drivers/clock_control/clock_control_mcux_syscon.c

```diff
@@ -76,18 +71,36 @@ static int mcux_lpc_syscon_clock_control_get_subsys_rate(
 	case MCUX_FLEXCOMM7_CLK:
 		*rate = CLOCK_GetFlexCommClkFreq(7);
 		break;
+	case MCUX_FLEXCOMM8_CLK:
+		*rate = CLOCK_GetFlexCommClkFreq(8);
+		break;
+	case MCUX_FLEXCOMM9_CLK:
+		*rate = CLOCK_GetFlexCommClkFreq(9);
+		break;
+	case MCUX_FLEXCOMM10_CLK:
+		*rate = CLOCK_GetFlexCommClkFreq(10);
+		break;
+	case MCUX_FLEXCOMM11_CLK:
+		*rate = CLOCK_GetFlexCommClkFreq(11);
+		break;
+	case MCUX_FLEXCOMM12_CLK:
+		*rate = CLOCK_GetFlexCommClkFreq(12);
+		break;
+	case MCUX_FLEXCOMM13_CLK:
+		*rate = CLOCK_GetFlexCommClkFreq(13);
+		break;
 	case MCUX_PMIC_I2C_CLK:
 		*rate = CLOCK_GetFlexCommClkFreq(15);
 		break;
 	case MCUX_HS_SPI_CLK:
-#if defined(FSL_FEATURE_FLEXCOMM8_SPI_INDEX)
+#if (FSL_FEATURE_SOC_FLEXCOMM_COUNT == 9)
```

Comment:
"9"? Bit of a magic number?

Reply:
Thanks for spotting this. I have changed the implementation to use HSLSPI register defines.